src/main/java/com/esri/samples/localserver/local_server_dynamic_workspace_raster/LocalServerDynamicWorkspaceRasterSample.java

```diff
@@ -0,0 +1,164 @@
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a control panel
+      VBox vBoxControl = new VBox(6);
+      vBoxControl.setMaxSize(240, 185);
+      vBoxControl.getStyleClass().add("panel-region");
+
+      // create the descriptive label
+      Label description = new Label("This application shows how to create a dynamic workspace connection " +
+              "to a raster folder and display raster data in a map. Click the " +
+              "button to select a local raster and add it to the map.");
+      description.setWrapText(true);
+
+      // create a file chooser to select a raster
+      FileChooser fileChooser = new FileChooser();
+      fileChooser.setTitle("Open Resource File");
+      fileChooser.getExtensionFilters().addAll(
+          new ExtensionFilter("Image Files", "*.tif"));
+      fileChooser.setInitialDirectory(new File("./samples-data/raster/"));
+
+      // create button to open file chooser
+      Button addButton = new Button("Choose Raster");
+      addButton.setMaxWidth(Double.MAX_VALUE);
+      addButton.setDisable(true);
+      addButton.setOnAction(e -> {
+       File rasterFile = fileChooser.showOpenDialog(stage);
+       if(rasterFile != null){
+         startLocalService(rasterFile);
+       }
+      });
+
+      // add button to the control panel
+      vBoxControl.getChildren().addAll(description, addButton);
+
+      // create a map and add it to a map view
+      ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // start the local server
+      server.startAsync().addDoneListener(() -> {
+        if (server.getStatus() == LocalServerStatus.STARTED) {
+          addButton.setDisable(false);
+        }
+      });
+
+      // add the map view and control panel to stack pane
+      stackPane.getChildren().addAll(mapView, vBoxControl);
+      StackPane.setAlignment(vBoxControl, Pos.TOP_LEFT);
+      StackPane.setMargin(vBoxControl, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Start a LocalMapService with the Raster File.
+   *
+   * @param rasterFile a raster file
+   */
+  private void startLocalService(File rasterFile) {
+    // start a service from a blank map package
+    String mapServiceURL = "./samples-data/local_server/mpk_blank.mpk";
+    LocalMapService localMapService = new LocalMapService(mapServiceURL);
+
+    // create a raster workspace
+    String fileName = rasterFile.getName();
+    String path = rasterFile.getParent();
+    RasterWorkspace rasterWorkspace = new RasterWorkspace("raster_wkspc", path);
+    RasterSublayerSource source = new RasterSublayerSource(rasterWorkspace.getId(), fileName);
+    ArcGISMapImageSublayer imageSublayer = new ArcGISMapImageSublayer(101, source);
```

Comment:
I would just start the id at 0. For some reason when I see 101 I start asking myself, is there a reason its 101, should we always start at 101, is 101 a certain key for a certain source layer. :)